Adding NUCLEO_WL55JC support

JeanMarcR opened the pull request:
<!--
For more information on the requirements for pull requests, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html).

NOTE: Do not remove any of the template headings (even for optional sections) as this
template is automatically parsed. 
-->

### Summary of changes <!-- Required -->

We would like to introduce the new STM32WL family.

Complementing the STM32 RF connectivity portfolio, the STM32WL System-On-Chip integrates both a general purpose microcontroller and a sub-GHz radio on the same chip.
Built on Arm® Cortex®‐M4 and Cortex®-M0+ cores (single- and dual-core architectures available - 48 MHz), STM32WL microcontrollers support multiple modulations– LoRa®, (G)FSK, (G)MSK, BPSK – to ensure flexibility in wireless applications with LoRaWAN®, Sigfox, W-MBUS or any other suitable protocol in a fully open way.
Thanks to a deep integration, the innovative and open architecture is optimized for LoRaWAN® legacy, Sigfox, proprietary protocols, flexible resource use, power management and helps lower BOM cost while offering a better user experience.

Developed using the same technology as the one implemented in our ultra-low-power STM32L4 microcontrollers, the STM32WL series provides similar digital and analog peripherals for basic or complex application use cases requiring an extended battery life and a long RF range through its sub-GHz transceiver.

To ensure worldwide compatibility, the STM32WL MCUs feature a dual power output and a wide linear frequency range fitting any unlicensed RF spectrum need.

Overall, the STM32WL series is the STM32 family’s pioneer in sub-GHz wireless connectivity, offering ease-of-use and reliability, while being perfectly tailored for a wide range of industrial and consumer applications.

Cube FW files are now available:
https://www.st.com/en/embedded-software/stm32cubewl.html

MBED-OS port has been implemented for NUCLEO_WL55JC target:

256 Kbytes of Flash memory
64 Kbytes of SRAM

https://www.st.com/en/microcontrollers-microprocessors/stm32wl55jc.html

 
### Impact of changes <!-- Optional -->
<!-- 
    If there are any implications for users taking this change then they must be 
    provided here. For Major PR types this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

#### Migration actions required <!-- Optional -->
<!-- 
    This should only be applicable in Major PR types for which this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

### Documentation <!-- Required -->

<!-- 
    Please provide details of any document updates required, including links to any
    related PRs against the docs repository.
    If no document updates are required please specify 'None', this at least tells us
    that this has been considered.
-->

----------------------------------------------------------------------------------------------------------------
### Pull request type <!-- Required -->

<!--
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front of them would change the meaning incorrectly. 
-->
    [x] Target update
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results <!-- Required -->

<!--
    Provide all the information required, listing all the testing performed. For new targets please attach full test results for all supported compilers.
-->
    [] No Tests required for this change (E.g docs only update)
    [x] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    Test executed on STMicroelectronics test bench with toolchain: ARM


    
----------------------------------------------------------------------------------------------------------------
### Reviewers <!-- Optional -->

<!--
    Request additional reviewers with @username or @team
-->

----------------------------------------------------------------------------------------------------------------

JeanMarcR committed:
STM32WL : ADDING STM32 SUPPORT

Add code concerning all STM32 platforms

JeanMarcR committed:
STM32WL : ADDING STM32WL SUPPORT

Add code concerning all STM32WL platforms
- STM32 Cube Firmware

Update CMakeLists.txt due to mbedtools evolution

JeanMarcR committed:
STM32WL : ADDING STM32WL55xC SUPPORT

Add code concerning all STM32WL55xC platforms
- Scatter loader and start-up files for
  ARM, GCC and IAR compilers.
- cmsis file
- Update CMakeLists.txt due to mbtools evolution

JeanMarcR committed:
STM32WL : ADDING NUCLEO STM32WL55JC SUPPORT

Add code concerning all STM32WL55JC platforms
- system clock, pin and peripheral definition
  mbedtools make file

Modify CmakeList to adapt to mbedtools evolution

JeanMarcR committed:
STM32WL : ADDING TARGET SUPPORT

Add STM32WL information in targets.json file

JeanMarcR committed:
STM32WL : ADDING ARM PACK MANAGER SUPPORT

Add STM32WL information in index.json file
Adding sector

jeromecoutant reviewed (approved):
NUCLEO_WL55JC is added in the ST CI test farm.
Results are OK

JeanMarcR committed:
STM32WL : ADDING CORTEX-M4 LINK OPTION

Add link option in file used for compilation.
Due to a bug in mbedtools environment.
